### PR TITLE
Mm 20576: : Migrate 'components/admin_console' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by ikeohachidi**
#### Summary
This PR migrates "components/admin_console" module and it's associated test to typescript

#### Ticket Link
  Jira Link: https://mattermost.atlassian.net/browse/MM-20576
  Fixes https://github.com/mattermost/mattermost-server/issues/13496

### CODE REVIEW

File: components/admin_console/index.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ import {getAdminDefinition} from 'selectors/admin_console';
 
 import LocalStorageStore from 'stores/local_storage_store';
 
-import AdminConsole from './admin_console.jsx';
+import AdminConsole from './admin_console';
```
[the remark this answers is not in the document]
Comment (ikeohachidi) [reply components/admin_console/index.js]: @devinbinnie Done.

File: package-lock.json [hunk 2]
```diff
@@ -5348,7 +5348,7 @@
         },
         "p-cancelable": {
           "version": "0.4.1",
-          "resolved": "http://registry.npmjs.org/p-cancelable/-/p-cancelable-0.4.1.tgz",
```
Comment (devinbinnie) [inline package-lock.json]: These changes to `package-lock.json` aren't required for this change, can we revert them?

File: components/admin_console/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (devinbinnie) [reply components/admin_console/index.js]: Looks pretty good, can you remove the old file and commit the new one as a rename?
It makes it MUCH easier for us to review. Thanks :)

File: package-lock.json (hunk 2 above)
Comment (ikeohachidi) [reply package-lock.json]: Oh, my mistake, I've reverted it

File: components/admin_console/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ikeohachidi) [reply components/admin_console/index.js]: I've done that

**Comment by ikeohachidi** [comment]
@saturninoabril I've fixed it.

**Comment by saturninoabril** [comment]
@ikeohachidi Thanks for fixing and for your contribution!

**Pull request merged**